Tchap/Modules/Room/Members/RoomParticipantsViewController.m

```diff
@@ -902,36 +902,39 @@ - (void)checkExpiredAccounts:(void (^)(void))completion
     dispatch_group_t requestsGroup = dispatch_group_create();
     self.userService = [[UserService alloc] initWithSession:self.mxRoom.mxSession];
     MXHTTPOperation *op;
-    for (Contact *contact in actualParticipants)
+    
+    NSArray<Contact*> *participants = [actualParticipants arrayByAddingObjectsFromArray:invitedParticipants];
+    NSMutableArray *contactIds = [NSMutableArray new];
+    for (Contact *contact in participants)
     {
         if (contact.mxMember.userId)
         {
-            dispatch_group_enter(requestsGroup);
-            op = [self.userService isAccountExpiredFor:contact.mxMember.userId
-                                          success:^(BOOL isExpired) {
-                                              contact.isExpired = isExpired;
-                                              dispatch_group_leave(requestsGroup);
-                                          } failure:^(NSError * _Nonnull error) {
-                                              contact.isExpired = false;
-                                              dispatch_group_leave(requestsGroup);
-                                          }];
+            [contactIds addObject:contact.mxMember.userId];
         }
     }
-    for (Contact *contact in invitedParticipants)
-    {
-        if (contact.mxMember.userId)
+    
+    dispatch_group_enter(requestsGroup);
+    op = [self.userService getUsersInfoFor:contactIds success:^(NSDictionary<NSString *,id> * _Nonnull usersInfo) {
+        for (Contact *contact in participants)
+        {
+            UserStatusInfo *userInfo = [usersInfo objectForKey:contact.mxMember.userId];
```

Comment:
`contact.mxMember` is nullable, you should check before using `contact.mxMember.userId` as a key

To get this kind of participant, invite an email (for which no account exists yet) in the room. Then `contact.mxMember` is null

Reply:
Applied